WKData/Sources/WKData/Data Controllers/Image Recommendations/WKImageRecommendationsDataController.swift

```diff
@@ -0,0 +1,42 @@
+import Foundation
+import UIKit
+
+public class WKImageRecommendationsDataController {
+
+	// MARK: - Nested Types
+
+	struct OnboardingStatus: Codable {
+		var hasPresentedOnboardingModal: Bool
+
+		static var `default`: OnboardingStatus {
+			return OnboardingStatus(hasPresentedOnboardingModal: false)
+		}
+	}
+
+	// MARK: - Properties
+
+	private let userDefaultsStore = WKDataEnvironment.current.userDefaultsStore
+
+	// MARK: - Lifecycle
+
+	public init() {
+
+	}
+
+	// MARK: - Onboarding
+
+	private var onboardingStatus: OnboardingStatus {
+		return (try? userDefaultsStore?.load(key: WKUserDefaultsKey.imageRecommendationsOnboarding.rawValue)) ?? OnboardingStatus.default
+	}
+
+	public var hasPresentedOnboardingModal: Bool {
```

Comment:
The user defaults store can save any Codable type, so you could just store a simple Bool without having to wrap it up in the `OnboardingStatus` struct:

```
    public var hasPresentedOnboardingModal: Bool {
        get {
            return (try? userDefaultsStore?.load(key: WKUserDefaultsKey.imageRecommendationsOnboarding.rawValue)) ?? false
        } set {
            try? userDefaultsStore?.save(key: WKUserDefaultsKey.imageRecommendationsOnboarding.rawValue, value: newValue)
        }
    }
```

The struct is fine if you plan on adding more stuff to it in the future, but if not I think this simplification is a little easier to read at a glance.

Reply:
Though not explicitly mentioned on the Phab task, I anticipated we'd potentially need to also track the progress of other feature-related onboarding activities (like whether the nice to have tooltips have displayed yet or not), but I'm happy to adjust it if that's not the case, or in whatever manner @mazevedofs feels is best for the feature.

Reply:
I think we can keep the struct since it paves the way for the tooltips implementation, we'll eventually get to implement the nice to haves in a future release.